### PR TITLE
Extract reusable parser tests

### PR DESCRIPTION
**Pull request by staugaard**
This moves the parser tests into a reusable module so that they can be used across multiple parsers.

This allows us to use the same tests on `graphql-libgraphqlparser-ruby`

### CODE REVIEW

**Comment by staugaard** [comment]
Huh? The tests pass just fine on my machine.

File: spec/graphql/language/parser_spec.rb [hunk 1]
```diff
@@ -1,562 +1,7 @@
 require "spec_helper"
+require 'graphql/Language/parser_tests'
```
Comment (rmosolgo) [inline spec/graphql/language/parser_spec.rb]: I think this should be a little `l`, not a big `L`! Maybe it passes on OSX where the filesystem is not case sensitive?

Comment (staugaard) [inline spec/graphql/language/parser_spec.rb]: Oh snap!

Comment (dphaener) [inline spec/graphql/language/parser_spec.rb]: Just to throw in my two cents, that's exactly the issue. Has bitten me quite a few times.

**Comment by staugaard** [comment]
All green now. Sorry about that.

**Comment by rmosolgo** [comment]
Awesome, thanks!

**Comment by rmosolgo** [comment]
:ship: in 0.18.6!

**Comment by staugaard** [comment]
Thanks!

**Pull request merged**